### PR TITLE
Make Twig 3.10.2 the minimum requirement

### PR DESCRIPTION
**Pull request by leofeyer**
This pull request makes Twig 3.10.2 the minimum version and therefore does not need all the BC layers.

@contao/developers Do we want to add this to Contao 5.3 or only in Contao 5.4?

### CODE REVIEW

**Comment by ausi** [comment]
> Do we want to add this to Contao 5.3

Yes.

File: core-bundle/src/Twig/Extension/ContaoExtension.php [hunk 1]
```diff
@@ -69,24 +69,19 @@ public function __construct(
...
-        $escaperExtension = $environment->getExtension(EscaperExtension::class);
 
-        // Forward compatibility with twig/twig >=3.10.0
-        if (method_exists($escaperExtension, 'setEnvironment')) {
-            $escaperExtension->setEnvironment($environment);
-        }
-
-        $escaperExtension->setEscaper('contao_html', $contaoEscaper->escapeHtml(...));
-        $escaperExtension->setEscaper('contao_html_attr', $contaoEscaper->escapeHtmlAttr(...));
+        $runtime = $this->environment->getRuntime(EscaperRuntime::class);
```
Comment (m-vo) [inline core-bundle/src/Twig/Extension/ContaoExtension.php]: Can we name the variable `$escaperRuntime` everywhere, so it is clear that we do not mean any of the other runtimes? ☺️

File: core-bundle/src/Twig/Extension/ContaoExtension.php [hunk 2]
```diff
@@ -242,39 +237,46 @@ function (Environment $env, $context, $template, $variables = [], $withContext =
 
     public function getFilters(): array
     {
-        $escaperFilter = static function (Environment $env, $string, $strategy = 'html', $charset = null, $autoescape = false) {
+        $escaperFilter = function (Environment $env, $string, string $strategy = 'html', string|null $charset = null, bool $autoescape = false) {
+            $runtime = $this->environment->getRuntime(EscaperRuntime::class);
```
Comment (ausi) [inline core-bundle/src/Twig/Extension/ContaoExtension.php]:
```suggestion
        $escaperFilter = static function (Environment $env, $string, string $strategy = 'html', string|null $charset = null, bool $autoescape = false) {
            $runtime = $env->getRuntime(EscaperRuntime::class);
```
I think we should use the `Environment` that gets passed to the filter here.

**Review by ausi — COMMENTED** [review]
One change, rest LGTM!

File: core-bundle/src/Twig/Extension/ContaoExtension.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Twig/Extension/ContaoExtension.php]: Changed in 6f7ccc2f4c60b862ab65cb0a8ff17ffc4432060c.

File: core-bundle/src/Twig/Extension/ContaoExtension.php (hunk 2 above)
Comment (leofeyer) [reply core-bundle/src/Twig/Extension/ContaoExtension.php]: Changed in 6f7ccc2f4c60b862ab65cb0a8ff17ffc4432060c.

**Review by m-vo — CHANGES_REQUESTED** [review]
We have a test `testContaoUsesCorrectTwigFunctionSignatures` that checks for the function signatures of the library functions. We now call the runtimes, but we still test the signature of the global functions. IMHO we should either change that or drop the test.

**Comment by leofeyer** [comment]
Dropped in 6dea4c84b9a30b8f67648da11491a29426a971d7.

**Review by m-vo — APPROVED** [review]

**Review by ausi — APPROVED** [review]
❤️

**Pull request merged**